src/router/useSimpleNavigation.js

```diff
@@ -62,11 +63,24 @@ function fetchRouteManifest() {
 }
 
 function getRoute(href, routes) {
+  const matches = [];
+
   for (let pattern in routes) {
     if (new RegExp(pattern, 'i').test(href)) {
-      return routes[pattern].as
+      matches.push(routes[pattern].as)
     }
   }
 
-  return null
+  // matches all occurrences of `...`, `[`, `]`, and `/` 
+  const specificityWeightPattern = new RegExp(/\.{3}|\[|\/|\]/g);
+
+  // sorts the matches by specificity weight descending
+  const prioritizedMatches = matches.sort((a, b) => {
+    const aWeight = (a.match(specificityWeightPattern) || []).length;
+    const bWeight = (b.match(specificityWeightPattern) || []).length;
+    return aWeight > bWeight ? 1 : -1
+  });
+
+  // return the route with the most specificity, similar to Next.js route matching
+  return prioritizedMatches[0] || null;
```

Comment:
This is the naive approach to prioritizing routes. It would be better to use a Next.js exposed util for this if one is available.